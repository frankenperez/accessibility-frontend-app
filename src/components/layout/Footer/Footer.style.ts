import styled from "@emotion/styled";

const StyledFooter = styled.footer`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${({ theme }) => theme.palette.grey[800]};
  height: 60px;
  padding: 10px 26px;

  .navigation {
    font-size: 14px;

    ul {
      display: flex;
      flex-direction: row;
      align-items: center;
      gap: 40px;
    }

    .link {
      position: relative;
      color: ${({ theme }) => theme.palette.primary.contrastText};
      text-decoration: none;

      &:after {
        content: "";
        position: absolute;
        width: 100%;
        transform: scaleX(0);
        height: 1px;
        bottom: 0;
        left: 0;
        background-color: ${({ theme }) => theme.palette.primary.contrastText};
        transform-origin: center center;
        transition: transform 0.25s ease-out;
      }

      &:hover::after {
        transform: scaleX(1);
        transform-origin: center center;
      }
    }
  }
`;

export default StyledFooter;
