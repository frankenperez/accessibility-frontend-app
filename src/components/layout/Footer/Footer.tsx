// React
import React from "react";

// Components
import NavLink from "components/common/NavLink/NavLink";

// i18n
import { useTranslation } from "react-i18next";

// Styles
import StyledFooter from "./Footer.style";

// Footer
export default function Footer(): JSX.Element {
  const { t } = useTranslation("app");
  // TODO: Get footerItems
  const footerItems = [
    {
      label: t("footer.contact"),
      url: "#"
    },
    {
      label: t("footer.privacy"),
      url: "#"
    },
    {
      label: t("footer.terms"),
      url: "#"
    }
  ];
  return (
    <StyledFooter>
      <nav className="navigation">
        <ul>
          {footerItems.map((link) => {
            const { label, url } = link;
            return (
              <li key={label}>
                <NavLink className="link" to={url} target="_blank" rel="noreferrer" activeClassName="active">
                  {label}
                </NavLink>
              </li>
            );
          })}
        </ul>
      </nav>
    </StyledFooter>
  );
}
