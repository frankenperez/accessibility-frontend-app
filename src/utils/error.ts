import axios from "axios";

export const errorParser = (error: unknown) => {
  let message: string;
  if (axios.isAxiosError(error)) {
    const errorData = error.response?.data as { reason: string } | string;
    if (typeof errorData === "object" && errorData?.reason) {
      message = errorData?.reason;
    } else {
      message = JSON.stringify(errorData);
    }
  } else if (error instanceof Error) {
    message = error.message;
  } else {
    message = JSON.stringify(error);
  }
  return message;
};
