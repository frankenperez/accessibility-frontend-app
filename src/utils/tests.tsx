// React
import React, { PropsWithChildren } from "react";

// Testing Library
import { render, RenderOptions } from "@testing-library/react";

// Router
import { BrowserRouter } from "react-router-dom";

// Internationalization
import en_BG from "../languages/en-GB.json";
import i18next from "i18next";
import { I18nextProvider } from "react-i18next";

// MUI Theme
import { createTheme, StyledEngineProvider, ThemeProvider } from "@mui/material/styles";
import theme from "styles/theme";
const appTheme = createTheme(theme);

// Internationalization config
i18next.init({
  interpolation: { escapeValue: false }, // React already does escaping
  lng: "en", // Language to use
  resources: {
    en: {
      app: en_BG
    }
  }
});

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  route?: string;
}

const renderWithProviders = (
  ui: React.ReactElement,
  {
    route = "/",
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren): JSX.Element => {
    return (
      <I18nextProvider i18n={i18next}>
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={appTheme}>
            <BrowserRouter>{children}</BrowserRouter>
          </ThemeProvider>
        </StyledEngineProvider>
      </I18nextProvider>
    );
  };
  window.history.pushState({}, "Test page", route);

  // Return an object with the store and all of RTL's query functions
  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders };
