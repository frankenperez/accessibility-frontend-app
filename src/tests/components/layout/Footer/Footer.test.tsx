// React
import React from "react";

// Component
import Footer from "components/layout/Footer/Footer";

// Test Utils
import { renderWithProviders } from "utils/tests";

// Tests
describe("Tests for <Footer />", () => {
  it("Renders with default props", () => {
    const { container } = renderWithProviders(<Footer />);
    // Creates the snapshot test
    expect(container).toMatchSnapshot();
  });
});
