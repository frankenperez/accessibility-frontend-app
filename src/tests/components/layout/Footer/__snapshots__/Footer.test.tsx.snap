// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <Footer /> Renders with default props 1`] = `
<div>
  <footer
    class="css-ik1bx0"
  >
    <nav
      class="navigation"
    >
      <ul>
        <li>
          <a
            aria-current="page"
            class="link active"
            href="/"
          >
            Contact Us
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="link active"
            href="/"
          >
            Privacy Policy
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            class="link active"
            href="/"
          >
            Terms of Service
          </a>
        </li>
      </ul>
    </nav>
  </footer>
</div>
`;
